client/src/components/DataFiles/DataFiles.js

```diff
@@ -55,6 +78,15 @@ const DataFilesSwitch = React.memo(() => {
           );
         }}
       />
+      <Route
+        path={`${path}/shared`}
+        render={() => {
+          dispatch({
+            type: 'DATA_FILES_CLEAR_PROJECT_SELECTION'
```

Comment:
Why is this action needed here? Can it be rolled into the `PROJECTS_GET_LISTING` saga and renamed? (if we ever have a case for doing checkbox selection of projects this could get confusing).

Reply:
I've moved the dispatch to the DataFilesProjectListing component's useEffect, since it is a concern that affects that component only.

However, I can't pull the actual reducer case out and put it in the Projects reducer. The Data Files main view pulls state.files.params.FileListing from the Datafiles reducer in order to display Breadcrumbs. If I don't send the dispatch and have that reducer clear that object, the breadcrumbs will show the stale, last viewed file system when you click on the Shared Workspaces nav route. (Like if you click on My Data and then click Shared Workspaces the breadcrumbs will show My Data for Shared Workspaces.) Do you have a suggestion that might improve this, short of a refactor of breadcrumbs?

Reply:
I think I was unclear, sorry- my suggestion was to keep the action but call it from the saga instead of the router. That way the UI only dispatches one action (through `useEffect` in the project listing) and in the future we don't have to remember that there are 2 places where state changes are triggered.

Reply:
Ahhhh i see. Ok.

Reply:
But do you think there's a danger with mixing the "Retrieve projects list" and clearing file listing params? For example, if we have a system selector dropdown, there will be a point at which the "Retrieve projects list" saga runs to retrieve those systems (perhaps during a modal) that may clear the params at an undesirable time such as when the modal itself is currently looking at the My Data system.

Reply:
Discussed with jake. A shared workspaces saga will trigger the other two sagas.